source/src/PascalCompiler/Grammar/PascalRules.h

```diff
@@ -8,214 +8,234 @@
-    {ConstExpr2,{CJ}, TreeConstructor},
-    {ConstExpr2,{EPSILON}, TreeConstructor},
-    {RatianalType1,{CI,RatianalType2}, RationalStart},
-    {RatianalType2,{SRSP,CI}, RationalEnd},
-    {RatianalType2,{EPSILON}, TreeConstructor},
-// VAR DONE
-    {VarSection,{RWV,VarSection1,VarSection2}, TreeConstructor},
-    {VarSection1,{IdList,SRCN,Type,SRSM}, VariableDeclarationAction},
-    {VarSection2,{VarSection1,VarSection2}, TreeConstructor},
-    {VarSection2,{EPSILON}, TreeConstructor},
-// ID LIST DONE
-    {IdList,{ID,IdList1}, FinalAppendList},
-    {IdList1,{SRCA,ID,IdList1}, AppendIdList},
-    {IdList1,{EPSILON}, InitIdList},
+    {Program, {RWPR, ID, SRSM, GlobalDeclaration, Program1}, TreeConstructor},
+    {Program1, {FunctionDeclaration, CompoundStatement, SRSP}, FinishMainFuntion},
+    {Program1, {ProcedureDeclaration, CompoundStatement, SRSP}, FinishMainFuntion},
+    {Program1, {CompoundStatement, SRSP}, FinishMainFuntion},
+
+// GLOBAL DECLARATION
+    {GlobalDeclaration, {EPSILON}, TreeConstructor},
+    {GlobalDeclaration, {GlobalDeclaration1, GlobalDeclaration}, TreeConstructor},
+    {GlobalDeclaration1, {LabelSection}, TreeConstructor},
+    {GlobalDeclaration1, {ConstSection}, TreeConstructor},
+    {GlobalDeclaration1, {TypeSection}, TreeConstructor},
+    {GlobalDeclaration1, {VarSection}, TreeConstructor},
+    {GlobalDeclaration1, {OperatorOverloadSection}, TreeConstructor},
+
+// LABEL SECTION (DONE)
+    {LabelSection, {RWLB, Label, LabelSection1, SRSM}, LabelRegistration},
+    {LabelSection1, {EPSILON}, TreeConstructor},
+    {LabelSection1, {SRCA, Label, LabelSection1}, LabelRegistration},
+    {Label, {ID}, LabelID},
+
+// CONST SECTION (NOT SUPPORTED)
+    {ConstSection, {RWCN, ConstDefinition, SRSM, ConstSection1}, TreeConstructor},
+    {ConstSection1, {EPSILON}, TreeConstructor},
+    {ConstSection1, {ConstDefinition, SRSM, ConstSection1}, TreeConstructor},
+    {ConstDefinition, {ID, OPEQ, Expression}, TreeConstructor}, // Expression => ConstExpression
+
+// TYPE SECTION (DONE)
+    {TypeSection, {RWTP, TypeDefinition, SRSM, TypeSection1}, TreeConstructor},
+    {TypeSection1, {EPSILON}, TreeConstructor},
+    {TypeSection1, {TypeDefinition, SRSM, TypeSection1}, TreeConstructor},
+    {TypeDefinition, {ID, OPEQ, Type}, TypeRegistrationAction},
+    {Type, {BuiltInType}, TypePropagation},
+    {Type, {Massive}, TypePropagation},
+    {Type, {EnumType}, TypePropagation},
+    {Type, {RecordType}, TypePropagation}, // TODO RECORD
+    {Type, {ID}, GetTypeAccordingToID},
+    {DefiniteType, {BuiltInType}, TypePropagation},
+    {DefiniteType, {ID}, GetTypeAccordingToID},
+    {BuiltInType, {RWINT}, IntegerPass},
+    {BuiltInType, {RWRAT}, RationalPass},
+    {BuiltInType, {RWCX}, ComplexPass},
+    {Massive, {RWAR, SRLB, Index, IndexList, SRRB, RWOF, DefiniteType}, ArrayCreationAction},
+    {IndexList, {EPSILON}, IndexListInit},
+    {IndexList, {SRCA, Index, IndexList}, IndexListAppendAtStart},
```

Comment:
Right now, it looks like it's not really an `IndexList`. It seems like it's 'IndexListTail' due to inlining of the IndexList's head into rule's body, that is described two lines above. Consider renaming current `IndexList` into `IndexListTail` and creating additional rule `IndexList`->`Index` `IndexListTail`.